src/transformStateWithClones.js

```diff
@@ -7,7 +7,31 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  const res = [];
+  let currentState = { ...state };
+
+  for (const action of actions) {
+    switch (action.type) {
+      case 'addProperties':
+        currentState = Object.assign({}, currentState, action.extraData);
+        break;
+
+      case 'removeProperties':
+        currentState = Object.assign({}, currentState);
+
+        for (const key of action.keysToRemove) {
+          delete currentState[key];
+        }
+        break;
+
+      case 'clear':
+        currentState = {};
+        break;
+    }
```

Comment:
Consider adding a default case in the switch statement to handle any unexpected action types. This can help with error handling and ensure that your function behaves correctly even if it receives an action with an unknown type.